internal/envoy/v3/listener.go

```diff
@@ -202,18 +204,44 @@ func (b *httpConnectionManagerBuilder) ConnectionShutdownGracePeriod(timeout tim
 }
 
 func (b *httpConnectionManagerBuilder) DefaultFilters() *httpConnectionManagerBuilder {
+
 	b.filters = append(b.filters,
 		&http.HttpFilter{
 			Name: wellknown.Gzip,
```

Comment:
Are the wellknown names no longer required? If not, we could remove them, or at least add a comment that they're not important.

Reply:
Chatting with folks in Envoy slack, the names aren't supposed to be required I don't think, but do fail a proto constraint if you don't include. I can add a comment.

Reply:
Just re-read the issue; I'm curious if it still works if you put an arbitrary name in there that's not the well-known name.

Reply:
The name is not relevant now that the typed configs are used, however, there are checks like this (https://github.com/projectcontour/contour/blob/9fb3eba838f628d05b1280d899eb57fedbb3ef31/internal/envoy/v3/listener.go#L282) which check for the filter and panic if not there. 

I personally dislike this check since it causes Contour to exit and are better verified with unit tests.

Reply:
@skriss I took out the wellknown names and now do a check on the typeurl.